### DISCUSSION TITLE
Two types of templateID

### DISCUSSION THREAD

**Discussion by PR3R00T** [projectdiscovery/nuclei#1352] [Q&A] [answered]
After a painful troubleshoot in my automation it turns out that depending on the scan ran two different Key IDs of "template ID " is given.

```json
{
  "template": "exposed-panels/mobileiron-login.yaml",
  "template-url": "https://github.com/projectdiscovery/nuclei-templates/blob/master/exposed-panels/mobileiron-login.yaml",
  "template-id": "mobileiron-login",
  "info": {
    "name": "MobileIron Login",
    "author": [
      "dhiyaneshdk",
      "dwisiswant0"
    ],
    "tags": [
      "panel"
    ],
    "reference": null,
    "severity": "info"
  },
  "type": "http",
  "host": "https://BLAH",
  "matched-at": "https://BLAH/mifs/c/d/android.html",
  "ip": "1.1.1.1",
  "timestamp": "2021-12-07T14:25:35.536400515Z",
  "curl-command": "curl -X 'GET' -d '' -H 'Accept: /' -H 'Accept-Language: en' -H 'User-Agent: Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/36.0.1985.67 Safari/537.36' 'https://BLAH/mifs/c/d/android.html'",
  "matcher-status": true
}
```
```json
{
  "templateID": "package-json",
  "info": {
    "name": "npm package.json disclosure",
    "author": [
      "geeknik",
      "afaq"
    ],
    "tags": [
      "config",
      "exposure"
    ],
    "description": "All npm packages contain a file, usually in the project root, called package.json - this file holds various metadata relevant to the project.",
    "reference": null,
    "severity": "info"
  },
  "type": "http",
  "host": "https://BLAH2",
  "matched": "https://BLAH2/package.json",
  "timestamp": "2021-12-07T16:32:44.975067149Z"
}
```

Notice the first one is template-id and the second is templateID. Can this be corrected to a standard. preferably templateID :)

**Comment by PR3R00T** [comment]
also the same for another key "matched" and "matched-at".

**Comment by ehsandeep** [comment, marked as answer]
@PR3R00T this is not a bug or issue, looks like you mixed output between two releases? we have added new fields in the last release and updated a few ones in 2.5.3, all the details are included on the release page, so make sure to check and confirm it, let me know if you still have any questions.

- [v2.5.3 changelog](https://github.com/projectdiscovery/nuclei/releases/tag/v2.5.3)
- [v2.5.4 changelog](https://github.com/projectdiscovery/nuclei/releases/tag/v2.5.4)

**Reply by PR3R00T** [reply]
Gah, thats frustrating, no worries, thanks. Il have to do some changes then. Not going to hold you to it, but do you plan on changing any other fields @ehsandeep ?

**Reply by ehsandeep** [reply]
We updated all the fields we wanted to keep the output uniform in `2.5.3` release, so we don't expect any new updates, but new fields we be added as we add a new feature, so always keep eyes on release changelog.
